storybookjs/storybook
# IMPROVE HMR error reporting, no refreshes needed for error recovery

Issue: https://github.com/storybookjs/storybook/issues/5822

## What I did
I fixed the HMR for error handling and improved the display of errors a bit

## How to test
- open any example
- do a code-change in a story file
- assert the codechange is applied without a refresh
- do a code-change with a syntax error
- assert there is a error message
- navigate to another story
- assert that the error message stays in view
- fix the syntax error
- assert that the error disappears and you will navigate to the correct story

also:
- throw an error in the storyFn
- assert that the error is displayed
- fix the error in the story
- assert that the story renders correctly without a refresh

Here's how the error look now:
![Screenshot 2019-06-04 at 21 55 09](https://user-images.githubusercontent.com/3070389/58909534-a7bfbd00-8713-11e9-9b8c-5d3624e9f98c.png)
![Screenshot 2019-06-04 at 21 54 58](https://user-images.githubusercontent.com/3070389/58909535-a8585380-8713-11e9-95fd-b6e7cc50a9c1.png)
